src/coreclr/jit/emitxarch.cpp

```diff
@@ -15844,17 +15844,22 @@ BYTE* emitter::emitOutputLJ(insGroup* ig, BYTE* dst, instrDesc* i)
         // For forward jumps, record the address of the distance value
         id->idjTemp.idjAddr = (dstOffs > srcOffs) ? dst : nullptr;
 
-        dst += emitOutputLong(dst, distVal);
+        // Cross jumps may not be encodable in a 32-bit displacement as the
+        // hot/cold code buffers may be allocated arbitrarily far away from
+        // each other. We simply encode a 0 under the assumption that the
+        // relocations will take care of it.
+        bool crossJump = emitJumpCrossHotColdBoundary(srcOffs, dstOffs);
```

Comment:
Nit: Maybe we can set `crossJump` earlier in the method and replace the other calls to `emitJumpCrossHotColdBoundary()` with it, just to get our money's worth.

Reply:
`emitJumpCrossHotColdBoundary` isn't computed unconditionally in a dominator, so it would mean always computing it. It looks like it is super cheap from the TP diffs, so I'll just leave it like this.